### PR TITLE
Clean up of documentation build

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

removes warnings and errors from documentation build:
- normalize section headings
- fixes incomplete table markup
- fixes ~~typoes~~ typos

#### Select relevant component teams: 

@codice/docs 

#### How should this be tested?

build normally. Log output reduced, content renders 

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 To avoid the complexities of working with the Filter interface directly and implementing the ${branding} Profile of the Filter specification, the ${ddf-catalog} includes an API, primarily in `${ddf-branding}.filter`, to build Filters using a fluent API.
 
-To use the `FilterBuilder` API, an instance of `${ddf-branding}.filter.FilterBuilder` should be used via the OSGi registry.
+To use the `FilterBuilder` API, an instance of `${ddf-branding}	.filter.FilterBuilder` should be used via the OSGi registry.
```
Comment (shaundmorris) [inline distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc:10]: ❔ So we needed an intentional space here?

**Review by shaundmorris — APPROVED** [review]

File: distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc (hunk 1 above)
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc:10]: nope. fixed

**Review by jrnorth — APPROVED** [review]

File: distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_architectures/_queries/filterbuilder-api.adoc]: there should not. I took it back out

**Comment by ricklarsen** [comment]
build now

**Pull request merged**